src/main/java/org/apache/mesos/executor/CustomExecutor.java

```diff
@@ -86,23 +63,12 @@ public void launchTask(final ExecutorDriver driver, final Protos.TaskInfo task)
         LOGGER.info("Launching task: {}", task);
 
         try {
-            if (!task.hasData()) {
-                throw new ExecutorTaskException("Task data is not specified. Nothing to do.");
-            }
+            final ExecutorTask taskToExecute = executorTaskFactory.createTask(task, driver);
 
-            final Protos.CommandInfo taskData = Protos.CommandInfo.parseFrom(task.getData());
-            final Protos.Environment environment = taskData.getEnvironment();
-            final Map<String, String> taskEnv = TaskUtils.fromEnvironmentToMap(environment);
-
-            if (!taskEnv.containsKey(DcosTaskConstants.TASK_TYPE)) {
-                throw new ExecutorTaskException("Unable to determine task type: " + taskEnv);
-            }
-
-            final String taskType = taskEnv.get(DcosTaskConstants.TASK_TYPE);
-            final ExecutorTask taskToExecute = executorTaskFactory.createTask(taskType, task, driver);
-            executorService.submit(taskToExecute);
-            scheduleHealthCheck(task, taskToExecute);
-            launchedTasks.put(task.getTaskId(), taskToExecute);
+            Future<?> future = executorService.submit(taskToExecute);
+            LaunchedTask launchedTask = new LaunchedTask(taskToExecute, future);
```

Comment:
I generally don't like having construction of an object imply starting or execution, butI could do that.  It seems a little weird to create something called `LaunchedTask` which at creation launches itself?